tests/test_client.py

```diff
@@ -1381,6 +1390,10 @@ def _do_retrieve_scopes_test_helper(self, response, content,
         assertUrisEqual(self, token_uri, called_uri)
         logger.info.assert_called_once_with('Refreshing scopes')
 
+        http_request.assert_called_once_with(
+            token_uri, method='GET', body=None, headers=None,
+            redirections=5, connection_type=None)
```

Comment:
@jonparrott I (think I) now see why you didn't make this assertion, the `token_uri` and `called_uri` might not be literally equal due to dictionary order in Python (flaky failed tests).

Fail by me for not noticing above

```python
self.assertEqual(len(http_request.mock_calls), 1)
scopes_call = http_request.mock_calls[0]
```